File: python_backend/ranges/specialMoves/castle/get_castle_target_square.py
from custom_except import *
from ranges.specialMoves.castle.castle_target_squares import castle_target_squares


def get_castle_target_square(color, side):
    """get the target square of the castle for the given color king and on given side of king"""
    return castle_target_squares[color][side]

if __name__ == "__main__":
    print(get_castle_target_square('W', 'Q'))    
    print(get_castle_target_square('W', 'K'))    
    print(get_castle_target_square('B', 'Q'))    
    print(get_castle_target_square('B', 'K'))

File: node.js/ranges/specialMoves/castle/getCastleTargetSquare.js
var  castleTargetSquares  = require("./castleTargetSquares");
const { rfToXy } = require("../../../coordType/crdCnvrt")

function getCastleTargetSquare(color, side) {
    /*get the target square of the castle for the given color king &&  on given side of king**/
    return rfToXy(castleTargetSquares[color][side])
}

module.exports = getCastleTargetSquare;

if (require.main === module) {
    console.log(getCastleTargetSquare('W', 'Q'))    
    console.log(getCastleTargetSquare('W', 'K'))    
    console.log(getCastleTargetSquare('B', 'Q'))    
    console.log(getCastleTargetSquare('B', 'K'))
}